File: DataStructure_TA_PondS/Data Structure Algorithm/Source Code/Chapter 5/C/NodeChar.cpp
#include <stdio.h>
#include <stdlib.h>

struct NodeChar{
  char item;
  struct NodeChar *next;
};

struct NodeChar *insertNode(char newItem,struct NodeChar *nextNode){

  struct NodeChar *newNode;

  newNode = (struct NodeChar *)malloc(sizeof(struct NodeChar));
  newNode->item = newItem;
  newNode->next = nextNode;
  return(newNode);

}

void printlist (struct NodeChar *head){
  struct NodeChar *curr;
  curr = head;
  while (curr != NULL){
    printf("%d\t",curr->item);
    curr = curr->next;
  }
}

File: DataStructure_TA_PondS/Data Structure Algorithm/Source Code/Chapter 5/Java/Chapter5/src/chapter5/NodeChar.java
package chapter5;

public class NodeChar {

    private char item;
    private NodeChar next;
    //begin constructor

    public NodeChar() {
        next = null;
    }

    public NodeChar(char newItem) {
        item = newItem;
        next = null;
    }

    public NodeChar(char newItem, NodeChar nextNode) {
        item = newItem;
        next = nextNode;
    }//end constructor	
    //begin method

    public void setItem(char newItem) {
        item = newItem;
    }

    public char getItem() {
        return item;
    }

    public void setNext(NodeChar nextNode) {
        next = nextNode;
    }

    public NodeChar getNext() {
        return next;
    }
}
